File: app/src/main/java/com/example/meyss/javaretrofitdagger/di/ActivityBuilderModule.java
package com.example.meyss.javaretrofitdagger.di;

import com.example.meyss.javaretrofitdagger.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
@Module
public abstract class ActivityBuilderModule {

    @SuppressWarnings("unused")
    @ContributesAndroidInjector
    abstract MainActivity mainActivity();

}

File: app/src/main/java/com/example/meyss/javaretrofitdagger/di/ActivityBuilderModule.kt
package com.example.meyss.javaretrofitdagger.di

import com.example.meyss.javaretrofitdagger.Activities.MainActivity
import com.example.meyss.javaretrofitdagger.Activities.PokDetails
import com.example.meyss.javaretrofitdagger.Fragments.AttacksFragment
import com.example.meyss.javaretrofitdagger.Fragments.PokDetailsFragment1

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuilderModule {

    @ContributesAndroidInjector//(modules = [FragmentBuilderModule::class])
     abstract fun mainActivity(): MainActivity


    @ContributesAndroidInjector(modules = [FragmentBuilderModule::class])
     abstract fun pokDetails(): PokDetails



}
